File: BackEnd.v2.1/demo/src/main/java/com/example/demo/models/Cable.java
package com.example.demo.models;


import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.UUID;

@Data
@Entity
@Table(name = "cable")
public class Cable {

    @Id
    @GeneratedValue
    private UUID id;
    private String model;
    private String type;
    private Float length;
    private Float diameter;
    private Float nominal_voltage;
    private String section_transversal;
    private String material;
    private String isolation;
    private Float resistance;
    private Float price;
    private Integer quantity;
    @Lob
    @Column(name = "image", length = 500000000)
    private String image;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Company company;
}

File: FrontEnd.v2/src/app/models/cable.ts
/* tslint:disable */

/* eslint-disable */
import {Company} from "./Company";

export interface Cable {
  companyDto: Company;
  diameter: number;
  id?: any;
  image: string;
  isolation: string;
  length: number;
  material: string;
  model: string;
  nominal_voltage: number;
  price: number;
  quantity: number;
  resistance: number;
  section_transversal: string;
  type: string;
}
